### DISCUSSION TITLE
"Editing a clip can move other clips" defaults to "off"

### DISCUSSION THREAD

**Discussion by SteveDaulton** [audacity/audacity#1851] [General]
**Describe the bug**
By default I should be able to generate audio or paste audio into a track without being told there isn't room

**To Reproduce**
1. Reset Audacity preferences
2. Observe that "Editing a clip can move other clips" is disabled.

**Expected behavior**
 I should be able to generate audio or paste audio into a track without being told there isn't room

**Additional information (please complete the following information):**
 - OS: Xubuntu 20.04
 - Version 3.1.0-alpha-20211006

**Additional context**
This is a regression.

**Comment by LWinterberg** [comment]
This is intentional. It's very evident why the error message appears and thanks to cliphandles, it now also is very evident how to work around it. 

The old default did end up breaking sync between tracks when editing on multiple tracks. We found that behavior to be way more annoying and so flipped the default.

**Comment by SteveDaulton** [comment]
> The old default did end up breaking sync between tracks when editing on multiple tracks.

It was also the way that VI users bring tracks into alignment when editing on multiple tracks.



> and thanks to cliphandles, it now also is very evident how to work around it.

Not so evident if the track has many clips and the user is unfamiliar with moving multiple clips together. Rather than simply pasting a clip at an arbitrary position in the track, they may now have to zoom out, move multiple clips out of the way, paste into the gap, move the clips back to close up excess space. The workaround for me is obvious, I can just change the preference setting, but not everyone will know how to  do that.

**Comment by SteveDaulton** [comment]
> The old default did end up breaking sync between tracks when editing on multiple tracks.

How does disallowing pasting / generating into a track help with that issue? If the user needs to insert audio into a track, they need to insert audio into a track, and making that more difficult does not keep multiple tracks synchronized.

If the concern is that "deleting" audio from a track sync between tracks, then why not address that directly and make "Split Delete" the default delete action (and "modifier + delete" for "old style" delete)?

**Reply by LWinterberg** [reply]
The use case is if you have a project like this: 
![image](https://user-images.githubusercontent.com/87814144/136354388-8dde5694-1ef0-4bcf-96c3-e6634f84753c.png)

and want to instert a short sound effect which would fit, it just shifts *just* this track and breaks sync: 
![image](https://user-images.githubusercontent.com/87814144/136354854-1831f099-1208-4454-89fb-5096e84e6556.png)

Ripple editing is useful, but iffy as the default, similar to how punch and roll record is useful, but would be iffy as a default. I'd argue that ripple editing affecting all tracks (moving all clips which start later than the current selection) would be very useful, too, but also not very good as a default.

FWIW: Audition follows the same default as what we do now, but they of course have a more elegant solution in that they already have the idea of "dragging clips over each other crossfades between them", so pasting it in at a place it doesn't fit causes a crossfade rather than an error. I think this is something we should try to get built ASAP as well, but not for 3.1

**Reply by SteveDaulton** [reply]
> and want to instert a short sound effect which would fit, it just shifts _just_ this track and breaks sync:

That seems a bit tenuous. 
* If the *sound effect* was imported, then it can simply be dragged into position without moving any clips - even easier now that clips have drag bars.
* If the *sound effect* is generated, then isn't it more common to select the regions that you want to generate into? (again, this does not move other clips)

Which I think leaves us with one (user error) case of pasting a clip at a point, which could be solved in other ways, such as:
* Having a "Paste Insert" command (insert into empty space without moving clips)
* A "don't tell me again" warning that other clips will move (I don't much like pop-up warnings, but we have one now with the "no space" message).
* Right click menu item: "Paste into space" (like "Paste Insert")
* (More "advanced" option) A command to select the length of the clipboard contents
* Other

Turning off "Editing a clip can move other clips" seems to me to be an *easy way out* of a perceived problem that in reality is only an issue for a small minority, at the expense of all other users, rather than actually addressing the UX issue.

**Comment by Paul-Licameli** [comment]
May I suggest that the error dialog that opens when pasting fails should have a “?” button or a mention of the relevant Preferences page or some other such clue.

**Reply by petersampsonaudacity** [reply]
> May I suggest that the error dialog that opens when pasting fails should have a “?” button or a mention of the relevant Preferences page or some other such clue.

@Paul-Licameli it already does - see: file:///C:/Program%20Files/Audacity%203.1.3/help/manual/man/error_insufficient_space_in_track.html
![image](https://user-images.githubusercontent.com/50205984/166901523-60fc67ee-ef1c-435d-ad13-653c5ee9f350.png)


**_BUT_** I note that the Manual is incorrect as that page is still written as though the default for "Editing a clip can move other clips" is the old "on" setting which is no longer the case.  I will fix this for the 3.2.0 alpha Manual.

**Comment by DavidBailes** [comment]
Some initial thoughts:
1. Given that there must be quite a few users who use Audacity for recording and editing a single track, for example recording audio books, having ripple editing on by default might be a good idea.
2. Whatever the chosen default, it should be easier to change the mode, for example on the tracks menu there could be options for ripple editing off, on, on all tracks. (Isn't ripple editing on all tracks fairly similar to sync-lock tracks?).
3. If the default mode is changed, users should be clearly told about this change.

**Reply by LWinterberg** [reply]
> Isn't ripple editing on all tracks fairly similar to sync-lock tracks?

Yes, although right now, with "editing a clip can move other clips" turned on, a sync-lock paste will generate silence in the middle of adjacent clips

> If the default mode is changed, users should be clearly told about this change.

This definitely is going in the "what's new" page in the manual, and also might go in the release video which will get featured in the update and first launch dialogs.

**Reply by lukebrowell** [reply]
The law of unintended consequences applies. 

Please see https://github.com/audacity/audacity/issues/2893

**Comment by petersampsonaudacity** [comment]
I have updated the 3.2.0 alpha Manual correcting this page for the fact the "Editing a clip can move other clips" is now "'''''off'''''" by default:
https://alphamanual.audacityteam.org/man/Error:_Insufficient_space_in_track

I hope I have also made it clearer.

**Comment by petersampsonaudacity** [comment]
From my researches on the Internet - general opinion appears to be that it is better if Ripple editing is turned off by default - i.e. the changed default setting for  "Editing a clip can move other clips" to "'''''off'''''" by default is to be preferred - with users opting in for Ripple editing (somehow)
1.  https://www.homebrewaudio.com/8727/what-is-ripple-editing/
2. https://ask.audio/articles/ripple-editing-in-reaper - off by default in Reaper
3. https://www.cakewalk.com/Documentation?product=SONAR&language=3&help=Arranging.22.html - off by default in Cakewalk
4. etc.

**Comment by Tantacrul** [comment]
When we implement the ability for clips to overlap, this is going to be much less of a problem. It has always been the plan, although RT effects has taken a bit longer to complete than we expected! Anyway, it's on the shortlist of the next things we want to do.

**Comment by developer-controls** [comment]
Per the exchange at https://github.com/audacity/audacity/issues/2723, LWinterberg says that option is going away in the next update (3.3), which I interpret as there will be no longer be any way to enable "Editing a clip can move other clips." Is that correct?

My chief concern is for macros. Currently, if that option is disabled, all my macros that adjust volume and involve adding and removing noise and silence from the ends to avoid distortion at the ends fail with the error, "There is not enough room available to paste the selection." If the option goes away, are there updated macro commands that will still allow the copy and paste macro operations to work?

Here's an example:

```
SelectTime:End="-0.5" RelativeTo="ProjectEnd" Start="0"
Silence:Use_Preset="<Current Settings>"
Select:End="0" Mode="Set" RelativeTo="ProjectEnd" Start="0.5"
Copy:
CursProjectStart:
Paste:
SelectTime:End="-0.5" RelativeTo="ProjectEnd" Start="0"
Noise:Amplitude="0.3" Type="White"
Select:End="0" Mode="Set" RelativeTo="ProjectEnd" Start="0.5"
Copy:
CursProjectStart:
Paste:
SelectAll:
Join:
Compress&dynamics:compress-ratio="0.85" floor="-32" hardness="0.6" noise-factor="2" scale-max="0.65" use-percep="0"
SelectTime:End="1" RelativeTo="ProjectStart" Start="0"
Delete:
SelectTime:End="0" RelativeTo="ProjectEnd" Start="1"
Delete:
CursProjectStart:
```

I know I can add a command to the start to set the option (SetPreference:Name="/GUI/EditClipCanMove" Reload="0" Value="1"), but if the option is gone in 3.3, will that still work?

**Reply by LWinterberg** [reply]
Again, overlapping clips mean that there'll never be not enough room.

**Reply by SteveDaulton** [reply]
So "ripple editing" will be impossible in Audacity 3.3?
